[#5450] subtask(web): support web ui for creating table (basic column type)

orenccl opened the pull request:
### Proposed Changes in This Pull Request

- Added support for a web UI to create relational tables (basic operations).
- Enhanced the table list view with options to edit properties and delete tables.

### Why Are These Changes Necessary?

These changes introduce the ability to create tables directly from the UI, improving user experience and efficiency.

Closes: #5450

### User-Facing Changes

- Added a "Create Table" button in the table list view.
- Introduced a dialog for creating and updating tables.
- Enabled actions for viewing, editing properties, and deleting tables from the table list view.

### How Was This Patch Tested?

1. Started the Gravitino playground.
2. Ran the web UI to test the create table functionality, as well as the view, edit properties, and delete actions.
3. Conducted a lint check using Prettier for code formatting.
4. Built the entire project to ensure no build errors.

![2024-11-08 002032](https://github.com/user-attachments/assets/004f7355-d0de-420a-922c-046a2b91d024)

![image](https://github.com/user-attachments/assets/16642425-adb7-4ca4-a9b7-283879dedb22)

![image](https://github.com/user-attachments/assets/ad90bab5-6a89-4ef9-8396-4b73d0f4dc99)

LauraXia123 commented on web/web/src/app/metalakes/metalake/rightContent/CreateTableDialog.js:
```diff
@@ -0,0 +1,643 @@
+   * Add a new empty column
+   */
+  const addColumn = () => {
+    const newColumn = { name: '', type: '', nullable: false, comment: '' }
```
nullable default value is 'true', but depending on the needs of the product, this field is used with `Not null`, and the default value of `Not null` will be `false`

orenccl replied:
Should I set default value depending on column type?

orenccl replied:
I set the default value to true to match the API behavior.
![image](https://github.com/user-attachments/assets/60c5a759-956f-49fd-b6fe-dbc66860ae6c)

LauraXia123 commented on web/web/src/app/metalakes/metalake/rightContent/CreateTableDialog.js:
```diff
@@ -0,0 +1,643 @@
+            }
+          )
+        } else {
+          const updates = genUpdates(initialTableData, tableData)
```
the function `genUpdates` have no diff logic of table columns.

orenccl replied:
Hi @LauraXia123 
I think I need some advice on handling column update behavior and GUI design. For example, if the original columns are:
- `first`, `long`, `nullable`
- `second`, `long`, `nullable`

And the new columns are:
- `second`, `long`, `nullable`

How can I tell whether the operation is a column name change or a column deletion?

I’ve thought of two solutions, both involving UI adjustments to avoid this issue:
1. Allow users to update one column at a time rather than making multiple modifications and then updating.
2. Modify the UI so that users can execute a sequence of commands and see the results as they expect (see example picture).

---

![image (1)](https://github.com/user-attachments/assets/7d409a26-4f5c-4935-b3cb-83df6981c604)

orenccl replied:
Thank you for your reply. I will fix it within two days.

LauraXia123 commented on web/web/src/app/metalakes/metalake/rightContent/CreateTableDialog.js:
```diff
@@ -0,0 +1,650 @@
+   * Add a new empty column
+   */
+  const addColumn = () => {
+    const newColumn = { name: '', type: '', nullable: true, comment: '' }
```
table column also need check `hasDuplicateColumnName`

orenccl replied:
Ok

LauraXia123 commented:
<img width="1561" alt="image" src="https://github.com/user-attachments/assets/a8b727dc-b1fa-486b-b2c8-6cb834441f05">
Failed to create, probably because the submitted data should be deleted `hasDuplicateName` field.

LauraXia123 commented:
<img width="969" alt="image" src="https://github.com/user-attachments/assets/c9406e87-3b70-4931-ae0f-7e2dadbcfadf">
<img width="974" alt="image" src="https://github.com/user-attachments/assets/7026b108-31aa-4be0-94af-32cce82dd3ca">
Must have a column by default. Then you can't delete when there is only one column left

orenccl commented:
@LauraXia123  

Apologies—I should have tested the table creation feature after the latest commit.

I've corrected the issue and added functionality to hide the delete button when there is only one column. Additionally, I set a default value for the column.

LauraXia123 commented on web/web/src/app/metalakes/metalake/rightContent/CreateTableDialog.js:
```diff
@@ -0,0 +1,688 @@
+  const handleDialogClose = () => {
+    reset()
+    setInnerProps([])
+    setTableColumns([])
```
```suggestion
    setTableColumns([{ name: '', type: '', nullable: true, comment: '' }])
```

LauraXia123 commented:
Thank you very much for your contribution. LGTM

orenccl commented:
Thank you very much for your time
I will take extra care in future PRs.
